File: yeonwoo/0708_to_0714_week2_robot_regulatory_program/dog_robot.js
const Robot = require('./robot')

/**
 * Class with new key and method added to Robot class
 * a class that assign additional keys and values to some robot especially 'dog robot'
 * dog robot is a robot that seems like dog (animal) and is able to move
 *
 * <p>
 *     inherit from robot
 *     additional key = this.barkPower
 *          Since all robots do not have the function of 'power', I added only the 'bark power' key value.
 * </p>
 *
 * @author Yeonwoo Choi
 * @see see robot
 */
class DogRobot extends Robot{
    /**
     * Class with new key and method added to Robot class
     *
     * @author Yeonwoo Choi
     */
    constructor () {
        super()
        this.barkPower = 0
    }

    /**
     *method that assigns value to key
     * @param id
     * @param kind
     * @param name
     * @param x
     * @param y
     * @param price
     * @param distance
     * @param barkPower
     */
    setValue (id, kind, name, x, y, price, distance, barkPower) {

        this.robotId = id
        this.robotKind = kind
        this.robotName = name
        this.robotLocationX = x
        this.robotLocationY = y
        this.robotPrice = price
        this.robotDistance = distance
        this.barkPower = `${this.robotKind}=${barkPower}`
    }
}

module.exports = DogRobot

File: inseung/week3_class_Robot/DogRobot.kt
class DogRobot : Robot () {
    override var model = ""
    override var distance = 10
    override var position = Coordinate2D(150, 150)
    override var price = 0
    override var ID = 0

    var barkPower = 10

    override fun show () {
        print("DogRobot         "); super.show(); print("barkPower = $barkPower \n")
    }

    fun barkStart () = println("Bark Power ${barkPower}으로 짖기 시작했습니다")
    fun barkStop () = println("Bark Power ${barkPower}으로 짖다가 멈추었습니다")
}